[Docs] Add user level documentation for Expressions (#11990)
KratosMultiphysics/Kratos: docs/pages/Kratos/Expressions/IOs/Domain_Size_Expression_IO.md

---
title: Domain Size Expression IO
keywords: 
tags: [domain size, condition, element, expression io]
sidebar: kratos_expressions
summary: 
---

## Introduction

> ```DomainSizeExpressionIO``` computes the domain sizes of each entitiy (such as ```Condition``` or ```Element```). This only supports ```ConditionExpression``` or ```ElementExression```. The domain sizes are computed by accessing underlying geometry's ```Geometry::DomainSize``` method of the entity.

Reviewer: ```suggestion
```DomainSizeExpressionIO``` computes the domain sizes (length, surface or volume, depending of the entity's dimension) of each entitiy (such as ```Condition``` or ```Element```) in the provided `ModelPart`. This only supports ```ConditionExpression``` or ```ElementExression```. Domain sizes are computed by accessing underlying geometry's ```Geometry::DomainSize``` method.
```

1. If the ```Condition``` or ```Element``` as a geometry representing a line, then this will compute an expression with line length for each entitiy.
2. If the ```Condition``` or ```Element``` as a geometry representing a surface, then this will compute an expression with surface area for each entitiy.
3. If the ```Condition``` or ```Element``` as a geometry representing a volume, then this will compute an expression with volume for each entitiy.

> The resulting expression will be always a scalar expression.

Reviewer: ```suggestion
The resulting expression is always a scalar expression.
```
